kobo/apps/openrosa/apps/logger/management/commands/clean_duplicated_submissions.py

```diff
@@ -39,99 +38,158 @@ def add_arguments(self, parser):
             help="Specify a XForm's `id_string` to clean up only this form",
         )
 
+        parser.add_argument(
+            '--delete-unique-uuids',
+            action='store_true',
+            default=False,
+            help='Delete duplicates with identical uuid',
+        )
+
     def handle(self, *args, **options):
         username = options['user']
         xform_id_string = options['xform']
+        self._delete_unique_uuids = options['delete_unique_uuids']
+        self._verbosity = options['verbosity']
 
-        # Retrieve all instances with the same `uuid`.
+        # Retrieve all instances with the same `xml_hash`.
         query = Instance.objects
         if xform_id_string:
             query = query.filter(xform__id_string=xform_id_string)
 
         if username:
             query = query.filter(xform__user__username=username)
 
-        query = query.values_list('uuid', flat=True)\
-            .annotate(count_uuid=Count('uuid'))\
-            .filter(count_uuid__gt=1)\
+        query = (
+            query.values_list('xml_hash', flat=True)
+            .annotate(count_xml_hash=Count('xml_hash'))
+            .filter(count_xml_hash__gt=1)
```

Comment:
As per our internal discussion, `query` should be on `uuid` (as it was before) because we will compare `xml_hash` later.